tests/notification/test_sites.py

```diff
@@ -31,42 +31,54 @@ def __exit__(self, *args):
 
 class SiteIDTest(TestCase):
 
+    def setUp(self):
+        super(SiteIDTest, self).setUp()
+        self.second_site = Site.objects.create(domain='generic_second_site')
+
     @override_settings()
     def test_on_site_respects_SITE_ID_setting(self):
+        site_1 = Site.objects.get(id=1)
+        site_2 = Site.objects.create()
         site_1_pages = 3
         site_2_pages = site_1_pages + 1
-        with TempSiteID(1):
-            site_2 = Site.objects.create()
-            index = 0
-            for i in range(site_1_pages):
-                EmailNotification.objects.create(name=index)
-                index += 1
-            for i in range(site_2_pages):
-                notification = EmailNotification.objects.create(name=index)
-                notification.sites.add(site_2)
-                index += 1
-            self.assertEqual(EmailNotification.objects.on_site().count(), site_1_pages + site_2_pages)
+
+        index = 0
+        for i in range(site_1_pages):
+            notification = EmailNotification.objects.create(name=index)
+            notification.sites.add(site_1)
+            index += 1
+        for i in range(site_2_pages):
+            notification = EmailNotification.objects.create(name=index)
+            notification.sites.add(site_2)
+            index += 1
+
+        with TempSiteID(site_1.id):
+            self.assertEqual(EmailNotification.objects.on_site().count(), site_1_pages)
 
         with TempSiteID(site_2.id):
             self.assertEqual(EmailNotification.objects.on_site().count(), site_2_pages)
 
     @override_settings()
-    def test_site_not_added_multiple_times_on_save(self):
+    def test_site_overriden_on_save(self):
```

Comment:
this is replacing the original function of the test, no? shouldn't save still be called twice?